### DISCUSSION TITLE
MQTT values coming in incorrectly from poolController

### DISCUSSION THREAD

**Discussion by Hect0rG** [tagyoureit/nodejs-poolController#570] [Q&A] [answered]
I am working on setting up a view on my Openhab sitemap so that I can see what the controller is doing without having to pull up the GUI. I noticed three values that may be coming in incorrectly from my aquarite t15 SWG via MQTT explorer. Has anyone encountered this happening? 

I know the generator is on however the status shows of "off" even though I see the green generating led on the chlorinator.

```
nixie-singlebody/state/chlorinators/1/aquaritet15
{"id":1,"isOn":"off"}
```

Then I see that the salt required on the pool controller says "Add 1.88 40lb bags of salt" but the MQTT field says 75.

```
nixie-singlebody/state/chlorinators/1/aquaritet15/saltRequired
{"saltRequired":75}
```

And finally, the status is always showing 0 I'm not sure if this one is just telling me the setup is correct or if it is supposed to indicate if the generator is on or off.

```
nixie-singlebody/state/chlorinators/1/aquaritet15/status
{"status":{"val":0,"name":"ok","desc":"Ok"}}
```

**Comment by rstrouse** [comment, marked as answer]
Lets start with the last two as these are pretty easy.  saltRequired is in pounds of salt so 75/40 = 1.875 rounded to 1.88 40 pound bags.  The status from the chlorinator is one of 7 potential status values that are reported by the chlorinator.  Not all cells report all the values so some actually report no flow when there is another issue.
```
0 = Ok
1 = Low Flow
2 = Low Salt
3 = Very Low Salt
4 = High Current
5 = Clean
6 = Low Voltage
7 = Low Temp
8 = Comm Lost (this isn't actually reported by the cell cuz it cannot tell us that is cannot speak to us)
```

For the isOn status I will have a look.  It is possible that we are not emitting this consistently or that the cell is not reporting its current output.  Does dashPanel show the green status light as on when the cell is generating?

**Reply by Hect0rG** [reply]
Yes it does have green led on when generating and gui on nodejs-poolController  shows correctly as well. Just not coming through on MQTT.

**Reply by rstrouse** [reply]
Thanks that is helpful.  I'll have a look.

**Reply by rstrouse** [reply]
Pull njsPC.  Turns out that this value was never being set on MQTT.  The determination is that the currentOutput will be > 0 if the chlorinator is on.  The MQTT bindings have been modified to set the isOn flag to on|off depending on the current output.

**Reply by Hect0rG** [reply]
awesome. It's all working now thanks. I was able to make those changes in my openhab setup and all looks great now! pulling njsPC worked for the on/off. Thanks for taking care of that :)
